### PR TITLE
#54 Added an icon to the current position of the bike

### PR DESCRIPTION
**Pull request by heroichornet**
At the moment it is very rudimentary implementation. The icon is not changeable through gui or configuration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,10 @@ Features
 Tutorials
 --------
 - [GPS Tracks Animation mit "GPX Animator" (Marcus Bersheim)](https://www.youtube.com/watch?v=AtcBVrbB6bg) :de:
+
+
+Credits
+--------
+Icons included in application and their source:
+
+* <div>Icons made by <a href="https://www.flaticon.com/authors/freepik" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>
```
Comment (McPringle) [inline README.md]: This is a Markdown file, please write this line in Markdown syntax.
https://guides.github.com/features/mastering-markdown/#syntax
Or take a look at line 53. There is a link, too.

File: src/main/java/sk/freemap/gpxAnimator/Renderer.java [hunk 2]
```diff
@@ -532,6 +532,29 @@ private Point2D drawMarker(final BufferedImage bi, final int frame) {
 		return point;
 	}
 
+	private void drawSimpleCircleOnGraphics2D(Point2D point, Graphics2D g2) {
```
Comment (McPringle) [inline src/main/java/sk/freemap/gpxAnimator/Renderer.java]: When writing new code or refactoring existing code, I would like to see parameters and variables be `final`.

File: src/main/java/sk/freemap/gpxAnimator/Renderer.java [hunk 3]
```diff
@@ -509,16 +511,14 @@ private Point2D drawMarker(final BufferedImage bi, final int frame) {
...
-							markerSize,
-							markerSize);
-					g2.setStroke(new BasicStroke(1f));
-					g2.fill(marker);
-					g2.setColor(Color.black);
-					g2.draw(marker);
+
+					try {
+						drawIconOnGraphics2D(g2, point);
+					} catch (IOException e) {
```
Comment (McPringle) [inline src/main/java/sk/freemap/gpxAnimator/Renderer.java]: You should mark `IOException e` as `final`.

File: src/main/java/sk/freemap/gpxAnimator/Renderer.java [hunk 4]
```diff
@@ -532,6 +532,29 @@ private Point2D drawMarker(final BufferedImage bi, final int frame) {
...
+				point.getY() - markerSize / 2.0,
+				markerSize,
+				markerSize);
+		g2.setStroke(new BasicStroke(1f));
+		g2.fill(marker);
+		g2.setColor(Color.black);
+		g2.draw(marker);
+	}
+
+	private void drawIconOnGraphics2D(Graphics2D g2, Point2D point) throws IOException {
```
Comment (McPringle) [inline src/main/java/sk/freemap/gpxAnimator/Renderer.java]: Again, `final` would be really nice. That is not mandatory and not a bug, but we like to see this to make sure, the parameters don't get new values assigned in the body of the method.

**Comment by heroichornet** [comment]
I will make the icon configurable when I get the time. Also, maybe there is a better bike icon as a standard. I didn't search extensively.

**Comment by heroichornet** [comment]
Had to do some quick forced pushed because of a typo in my commit comment..

**Comment by McPringle** [comment]
Thank you very much for your awesome work! First, big step done!

**Pull request merged**